esf.tf

```diff
@@ -170,6 +170,9 @@ resource "aws_s3_object" "config-file" {
 }
 
 resource "terraform_data" "curl-dependencies-zip" {
+  // make the operation dependent on the release version
+  triggers_replace = [var.release-version]
+
```

Comment:
You can remove this, this is implicit already in the file name, since it depends on the version:

https://github.com/elastic/terraform-elastic-esf/blob/60f8797ba79516700476b91a17f939a9c367ce99/esf.tf#L11

Reply:
@constanca-m this is how we retrigger `terraform_data` for curl download if the version change. If we do not define this, Curl download will not retrigger and will need manual download of the version zip. 

See this - https://developer.hashicorp.com/terraform/language/resources/terraform-data#arguments

Reply:
I didn't really believe it, so I tried it with:

```terraform
locals {
  ecf_version = "2.0"
}

resource "terraform_data" "curl-dependencies-zip" {
  provisioner "local-exec" {
    command = "echo ${local.ecf_version}"
  }
}
```

1st try: echo printed
2nd try: not printed
3rd try: changed local variable, not printed again
4th try: added a triggers_replace, echo printed
5th try: not printed again (didn't change the variable)
6th try: changed the variable again to test the trigger, echo printed

Verdict: you are right! :wink: